Room Logic and Game screen rendering done

Wasif4life786 opened the pull request:
Checks JSON file for which Rooms are Traversable, creates a Map based on this using World Map Class. Implemented Logic to handle a player Transitioning from one room to another in Screen class. Screen class itself now displays the Game to declutter the main. In terms of displaying the Game Screen, all that's left is the background maps and other sprites.

Lordimass committed:
Added Main Path Map Datapoints

Added datapoints to `map.json` for each of the main path rooms.

Lordimass committed:
OutsideHub rooms

Lordimass committed:
LakesideWay rooms

Lordimass committed:
AnneLister Rooms

Lordimass committed:
DavidKato Rooms

Lordimass committed:
Top Half Of Map

Lordimass committed:
Law & RCH

Lordimass committed:
CSBuilding

Lordimass committed:
Merge branch 'main' into map-data

Wasif4life786 committed:
Loads the rooms into a Map

Room Class will load the JSON Map into a Java Map with each direction, also setup a convenience method that returns the next room

Lordimass committed:
Piazza

Lordimass committed:
LakePath

Lordimass committed:
LakePath

Lordimass committed:
Merge branch 'map-data' of https://github.com/WHALES-ENG1/WHALES-Game into map-data

Wasif4life786 committed:
Loading World Map from JSON file

Uses the Room Class to load the World Map. Also implemented a Method that should change the Location of the character depending on where they're allowed to move

Wasif4life786 committed:
Room Directional Logic implemented

Shifted location of JSON file logic into worldMap, also implemented screen logic to handle transition from one room into another. Modified Main to Check Room Transitions.

Wasif4life786 committed:
Moved Game Screen render logic

Moved Game Screen logic from main to Screen, this declutters the main and will be very useful when implementing the other screens. Temporarily removed inheritance from other screens.

Wasif4life786 committed:
Merge branch 'map-data' of https://github.com/WHALES-ENG1/WHALES-Game into map-data

Lordimass committed:
Restored `Map` Class

Lordimass committed:
Documented WorldMap

Lordimass committed:
Converted Room to use Enum

Lordimass reviewed (commented):
I'm gonna make a few changes and tidy a few bits up tomorrow but otherwise looks good.

Main thing I'd say is the lack of JavaDoc makes it quite hard to review and figure out what's going on, so that would be appreciated if possible please and thank you :)

Lordimass committed:
Restored Screen extensions to LibGDX Screen

using LibGDX ScreenAdapter instead of a custom Screen class since it already exists and can save us time.

Lordimass commented:
Realised these commits are not in the right branch, so I've cherry picked them over: https://github.com/WHALES-ENG1/WHALES-Game/tree/scene-changing

Also opened a new PR identical to this one from that branch instead: #47
